src/ListMessageSplitter.ts

```diff
@@ -34,35 +93,56 @@ export class MessageListItem {
     }
 }
 
+// A "header" object to push into the list splitter.
 export class MessageListHeader extends MessageListItem {
     public size(): number {
         return this.html.length + this.text.length
     }
 }
 
+// An internal helper class to hold a series of items, together with an optional header.
+//
+// Mainly provides a coherent split_at_size function that would allow for on-demand-sized splitting of
+// listings with headers.
 class MessageListing {
     public items: MessageListItem[] = [];
 
     constructor(public header: MessageListHeader | null) {
     }
 
+    // Attempts to split this listing into a `sized` and `rest` listing.
+    //
+    // Returns:
+    // `sized` != null, if sized was adequate, or had to be split
+    // `rest` != null, if sized was not adequate (and had to be split),
+    //   or the first item is too big to be split at the desired size.
     public split_at_size(desired_size: number): {sized: MessageListing | null, rest: MessageListing | null} {
         if (this.size() <= desired_size) {
+            // If the current listing is under the desired size, just return it, rest = null.
             return {sized: this, rest: null};
         } else {
+            // Else, split it.
+
+            // Create a new listing with just the current header,
+            // as a candidate for the new sized listing.
             let sized = new MessageListing(this.header);
             if (sized.size() > desired_size) {
                 // If the header alone is too much, just give up.
                 return {sized: null, rest: this}
             }
 
+            // Create another listing, and dump the rest of all items in there.
             let rest = new MessageListing(null);
+            // Be sure to shallow copy, to not disturb the `this` listing.
             rest.items = [...this.items];
 
+            // Perform initial shift of item into a variable.
             let current_item = rest.items.shift();
 
+            // Keep looping while there are still items left.
+            // (Replaced at the end of the loop)
             while (current_item !== undefined) {
-                // Add the new item tentatively
+                // Add the new item to `sized` tentatively.
```

Comment:
no, because there might be things about a listing's size that are optimised or changed as result of an item being added.

For example, if `sized` has no items in it, it will not emit a `<ul></ul>`, reducing 8 bytes of the final size. If it does have an item, it'll add that and the item.

So it is safer to try, and then back off to the last version of the listing if it becomes too big.

Reply:
Ohh, I see what you mean. That is quite unfortunate .

Reply:
I also don't want to duplicate logic to calculate it, as that's prone to desync if the original logic is updated.